### PR TITLE
update SQL diagnotics document

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

This PR:
+ updates all `SQL diagnosis` to `SQL diagnostics` or `SQL diagnostics` in all SQL diagnotics-related documents.
+ aligns parts of https://github.com/pingcap/docs-cn/pull/3154/ https://github.com/pingcap/docs-cn/pull/3330 https://github.com/pingcap/docs-cn/pull/3368/ that only update `system-table-sql-diagnosis.md`.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3154/ https://github.com/pingcap/docs-cn/pull/3330 https://github.com/pingcap/docs-cn/pull/3368/
- Other reference link(s): https://github.com/pingcap/docs/pull/2832

### CODE REVIEW

**Comment by yikeke** [comment]
Please help review this PR~ @lilin90

File: system-tables/system-table-sql-diagnostics.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+summary: Understand SQL diagnostics in TiDB.
+category: reference
+aliases: ['/docs/dev/reference/system-databases/sql-diagnosis/','/docs/dev/system-tables/system-table-sql-diagnosis/']
+---
+
+# SQL Diagnostics
+
+> **Note:**
+>
+> This is an experimental feature. It is recommended that you **DO NOT** use this feature in the production environment.
```
Comment (lilin90) [inline system-tables/system-table-sql-diagnostics.md:12]:
```suggestion
> **Warning:**
>
> SQL diagnostics is still an experimental feature. It is **NOT** recommended that you use it in the production environment.
```

**Comment by lilin90** [comment]
@TomShawn Please resolve the conflicts.

**Comment by TomShawn** [comment]
@lilin90 Conflicts resolve, PTAL again, thanks!

**Pull request merged**